autosklearn/data/validation.py

```diff
@@ -533,3 +538,34 @@ def join_and_check(
                              "Pandas DataFrames, numpy arrays, scipy csr  or  python lists. "
                              "Kindly cast your targets to a supported type."
                              )
+
+    def _is_data_valid_for_classification(self, y: np.ndarray) -> None:
```

Comment:
How can we actually have this function triggered? If `type_of_target` detects a continuous dataset in the classification part in automl.py, it would stop Auto-sklearn. Therefore, this appears to do double work?

Reply:
Yes, I agree type_of_target is a super set of checks, except in the case of multi label classification, where the output is interpreted as multiclass-multioutput


I can think of 2 ways to go:
1. we don't support anything that type_of_target doesn't support and request the user to comply with it. In other words, we do not encode nor handle multilabel targets of string type. This will be the cleanest solution and we align to what scikit learn support. In this example, we will have to manually encode the target.

2. We add a code to handle this specific multilabel scenario. If the target is multidimensional, we check if dtype is encodable we proceed. So this _is_data_valid_for_classification was an attempt to go with this approach, and I realize the name is confusing. It should be something like handle multilabel and have more in-depth checks.

Please let me know your preferences